[icu] build failure

### Operating system

Macos

### Compiler

Clang

### Steps to reproduce the behavior

```Shell
$VCPKG_ROOT/vcpkg install icu
```


### Failure logs

 % $VCPKG_ROOT/vcpkg install icu                  
Computing installation plan...
The following packages will be built and installed:
    icu[core,tools]:arm64-osx@74.2#1
Detecting compiler hash for triplet arm64-osx...
Restored 0 package(s) from /Users/petr/.cache/vcpkg/archives in 4.75 us. Use --debug to see more details.
Installing 1/1 icu[core,tools]:arm64-osx@74.2#1...
Building icu[core,tools]:arm64-osx@74.2#1...
-- Using cached icu4c-74_2-src.tgz.
-- Cleaning sources at /Users/petr/oss/vcpkg/buildtrees/icu/src/c-74_2-src-83138cd994.clean. Use --editable to skip cleaning for the packages you specify.
-- Extracting source /Users/petr/oss/vcpkg/downloads/icu4c-74_2-src.tgz
-- Applying patch disable-escapestr-tool.patch
-- Applying patch remove-MD-from-configure.patch
-- Applying patch fix_parallel_build_on_windows.patch
-- Applying patch fix-extra.patch
-- Applying patch mingw-dll-install.patch
-- Applying patch disable-static-prefix.patch
-- Applying patch fix-win-build.patch
-- Applying patch vcpkg-cross-data.patch
-- Applying patch darwin-rpath.patch
-- Using source at /Users/petr/oss/vcpkg/buildtrees/icu/src/c-74_2-src-83138cd994.clean
-- Getting CMake variables for arm64-osx-dbg
-- Getting CMake variables for arm64-osx-rel
-- Generating configure for arm64-osx
-- Finished generating configure for arm64-osx
-- Configuring arm64-osx-dbg
CMake Error at scripts/cmake/vcpkg_execute_required_process.cmake:112 (message):
    Command failed: /bin/bash -c "V=1 ./../src/c-74_2-src-83138cd994.clean/source/configure  \"--enable-tools\" \"--disable-samples\" \"--disable-tests\" \"--disable-layoutex\" \"--disable-silent-rules\" \"--verbose\" \"--disable-shared\" \"--enable-static\" \"--enable-debug\" \"--disable-release\" \"--prefix=/Users/petr/oss/vcpkg/installed/arm64-osx/debug\" \"--bindir=\\${prefix}/../tools/icu/debug/bin\" \"--sbindir=\\${prefix}/../tools/icu/debug/sbin\" \"--libdir=\\${prefix}/lib\" \"--includedir=\\${prefix}/../include\" \"--datarootdir=\\${prefix}/share/icu\""
    Working Directory: /Users/petr/oss/vcpkg/buildtrees/icu/arm64-osx-dbg
    Error code: 1
    See logs for more information:
      /Users/petr/oss/vcpkg/buildtrees/icu/config-arm64-osx-dbg-config.log
      /Users/petr/oss/vcpkg/buildtrees/icu/config-arm64-osx-dbg-out.log
      /Users/petr/oss/vcpkg/buildtrees/icu/config-arm64-osx-dbg-err.log

Call Stack (most recent call first):
  scripts/cmake/vcpkg_configure_make.cmake:863 (vcpkg_execute_required_process)
  ports/icu/portfile.cmake:66 (vcpkg_configure_make)
  scripts/ports.cmake:175 (include)


error: building icu:arm64-osx failed with: BUILD_FAILED
Elapsed time to handle icu:arm64-osx: 3.9 s
Please ensure you're using the latest port files with `git pull` and `vcpkg update`.
Then check for known issues at:
  https://github.com/microsoft/vcpkg/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+icu
You can submit a new issue at:
  https://github.com/microsoft/vcpkg/issues/new?title=[icu]+Build+error+on+arm64-osx&body=Copy+issue+body+from+%2FUsers%2Fpetr%2Foss%2Fvcpkg%2Finstalled%2Fvcpkg%2Fissue_body.md

The most interesting line from the `config-arm64-osx-dbg` log is:

`autoconf-archive is missing`

Which is actually the problem.

### Additional context

I have solved this by installing autoconf-archive via brew `brew install autoconf-archive`, however, I'm not sure whether this package should not be installed via vcpkg instead, because this basically means that I need to install a build dependency explicitly.

Also, maybe vcpkg should hint to do this even before attempting to configure the package?

The issue did reproduce when I uninstalled `autoconf-archive`, since `vcpkg_configure_make` requires it:
```
vcpkg@vcpkgs-Virtual-Machine vcpkg % brew uninstall autoconf-archive
Uninstalling /opt/homebrew/Cellar/autoconf-archive/2023.02.20... (600 files, 3.6MB)
vcpkg@vcpkgs-Virtual-Machine vcpkg %
vcpkg@vcpkgs-Virtual-Machine vcpkg %
vcpkg@vcpkgs-Virtual-Machine vcpkg % vim  /Users/vcpkg/Monica/vcpkg/buildtrees/icu/config-arm64-osx-dbg-err.log
configure: WARNING: unrecognized options: --disable-silent-rules
configure: error: 'autoconf-archive' is missing
```

```
vcpkg_configure_make(
    SOURCE_PATH "${SOURCE_PATH}"
    PROJECT_SUBPATH source
    AUTOCONFIG
    DETERMINE_BUILD_TRIPLET
    ADDITIONAL_MSYS_PACKAGES autoconf-archive
    OPTIONS
        ${CONFIGURE_OPTIONS}
        --disable-samples
        --disable-tests
        --disable-layoutex
    OPTIONS_RELEASE
        --disable-debug
        --enable-release
    OPTIONS_DEBUG
        --enable-debug
        --disable-release
)
```

--- a/ports/icu/portfile.cmake
+++ b/ports/icu/portfile.cmake
@@ -1,16 +1,16 @@
 if(NOT VCPKG_TARGET_IS_WINDOWS)
     message(WARNING "${PORT} currently requires the following programs from the system package manager:
-    autoconf-archive
+    autoconf automake autoconf-archive
 On Debian and Ubuntu derivatives:
-    sudo apt-get install autoconf-archive
+    sudo apt-get install autoconf automake autoconf-archive
 On recent Red Hat and Fedora derivatives:
-    sudo dnf install autoconf-archive
+    sudo dnf install autoconf automake autoconf-archive
 On Arch Linux and derivatives:
-    sudo pacman -S autoconf-archive
+    sudo pacman -S autoconf automake autoconf-archive
 On Alpine:
-    apk add autoconf-archive
+    apk add autoconf automake autoconf-archive
 On macOS:
-    brew install autoconf-archive\n")
+    brew install autoconf automake autoconf-archive\n")
 endif()
 
 string(REGEX MATCH "^[0-9]*" ICU_VERSION_MAJOR "${VERSION}")

--- a/ports/icu/portfile.cmake
+++ b/ports/icu/portfile.cmake
@@ -1,3 +1,18 @@
+if(NOT VCPKG_HOST_IS_WINDOWS)
+    message(WARNING "${PORT} currently requires the following programs from the system package manager:
+    autoconf-archive
+On Debian and Ubuntu derivatives:
+    sudo apt-get install autoconf-archive
+On recent Red Hat and Fedora derivatives:
+    sudo dnf install autoconf-archive
+On Arch Linux and derivatives:
+    sudo pacman -S autoconf-archive
+On Alpine:
+    apk add autoconf-archive
+On macOS:
+    brew install autoconf-archive\n")
+endif()
+
 string(REGEX MATCH "^[0-9]*" ICU_VERSION_MAJOR "${VERSION}")
 string(REPLACE "." "_" VERSION2 "${VERSION}")
 string(REPLACE "." "-" VERSION3 "${VERSION}")

--- a/ports/icu/portfile.cmake
+++ b/ports/icu/portfile.cmake
@@ -1,4 +1,4 @@
-if(NOT VCPKG_HOST_IS_WINDOWS)
+if(NOT VCPKG_TARGET_IS_WINDOWS)
     message(WARNING "${PORT} currently requires the following programs from the system package manager:
     autoconf-archive
 On Debian and Ubuntu derivatives:

--- a/ports/icu/vcpkg.json
+++ b/ports/icu/vcpkg.json
@@ -1,7 +1,7 @@
 {
   "name": "icu",
   "version": "74.2",
-  "port-version": 1,
+  "port-version": 2,
   "description": "Mature and widely used Unicode and localization library.",
   "homepage": "https://icu.unicode.org/home",
   "license": "ICU",

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -3450,7 +3450,7 @@
     },
     "icu": {
       "baseline": "74.2",
-      "port-version": 1
+      "port-version": 2
     },
     "ideviceinstaller": {
       "baseline": "2023-07-21",

--- a/versions/i-/icu.json
+++ b/versions/i-/icu.json
@@ -1,7 +1,7 @@
 {
   "versions": [
     {
-      "git-tree": "e5fa480ab4b85f1e42e72262027c96c7dd5ae0af",
+      "git-tree": "a36199768fe1f9676b61d5d28ed9ac36c9f041c0",
       "version": "74.2",
       "port-version": 2
     },

--- a/versions/i-/icu.json
+++ b/versions/i-/icu.json
@@ -1,7 +1,7 @@
 {
   "versions": [
     {
-      "git-tree": "ac34dd85b6ed88c8be1f835a7af7421c33b66ff7",
+      "git-tree": "e5fa480ab4b85f1e42e72262027c96c7dd5ae0af",
       "version": "74.2",
       "port-version": 2
     },

--- a/versions/i-/icu.json
+++ b/versions/i-/icu.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "ac34dd85b6ed88c8be1f835a7af7421c33b66ff7",
+      "version": "74.2",
+      "port-version": 2
+    },
+    {
       "git-tree": "dd30ff59dbfac290a1547dccc5ba44646fffa018",
       "version": "74.2",
       "port-version": 1

